types/enzyme/enzyme-tests.tsx

```diff
@@ -859,6 +865,14 @@ function ReactWrapperTest() {
         reactWrapper = new ReactWrapper<MyComponentProps, MyComponentState>(<MyComponent stringProp="1" numberProp={1} />, undefined, { attachTo: document.createElement('div') });
         reactWrapper = new ReactWrapper<MyComponentProps, MyComponentState>(<MyComponent stringProp="1" numberProp={1} />, reactWrapper, { attachTo: document.createElement('div') });
     }
+
+    function test_component_type() {
+      const wrapper1 = shallow(<div><ComponentType stringProp={"S"} numberProp={1} /></div>);
+      wrapper1.find(ComponentType);
```

Comment:
If I'm not mistaken this test could have passed before, since passing in the union type would have just devolved to matching the `EnzymePropSelector` overload. A good test here would be to tack on prop access and then assert on the result type, to ensure we're actually matching the more useful overload we expect:
`wrapper1.find(ComponentType).props().stringProp; // $ExpectType string`

Reply:
It's fixed